src/CPM/PertMain2.py

```diff
@@ -350,10 +482,342 @@ def getAllAlternativePaths(self, startActivity, endActivity, path=[]):
+    return subpathsSetRed
+
+  def returnPathSymbolic(self, path):
+    """
+      Method designed to print the symbolic name of a path
+      @ In, path, list, list of activities
+      @ Out, None
+    """
+    symbPath = []
+    for act in path:
+      symbPath.append(act.name)
+    return symbPath
+
+  def setActivitiesAbsTimes(self):
+    """
+      Method designed to assign, to each activity, its initial and final absolute time values
+      @ In, None
+      @ Out, None
+    """
+    for act in self.forwardDict:
+      Tin = self.startTime + datetime.timedelta(hours=self.infoDict[act]['es'])
+      Tfin = Tin + datetime.timedelta(hours=act.returnDuration())
+      act.setTimes(Tin,Tfin)
+
+  def returnScheduleEndTime(self):
+    """
+      Method designed to return the absolute end time of the aschdule
+      @ In, None
+      @ Out, endTime, float, absolute end time of the aschdule
+    """
+    startTime, endTime = self.getCriticalPath()[-1].returnAbsTimes()
+    return endTime
+
+  def saveScheduleToJsn(self, nameFile=None):
+    """
+      Method designed to print on file schedule in json format
+      @ In, nameFile, string, name of the generated file
+      @ Out, file in json format
+    """
+    if nameFile is None:
+      nameFile = 'schedule.json'
+
+    with open(nameFile, 'w', encoding="utf-8") as fp:
+      for act in self.forwardDict.keys():
+        json.dump(act.printToJsn(), fp, sort_keys=True, indent=4)
+        fp.write("\n")
+
+  def resourcesTemporalCheck(self):
+    self.reqResources = pd.DataFrame().reindex_like(self.resources)
+    self.reqResources = self.reqResources.replace(np.nan, 0)
+    for act in self.forwardDict:
+      absTimeVals = act.returnAbsTimes()
+      res = act.returnResources()
+      if res is not None:
+        self.reqResources.loc[absTimeVals[0]:absTimeVals[1],res] += 1
+
+
+'''  def getSubpathsParalleltoCP(self, CP, paths):
+    subpaths = []
+
+    pathsOrdered = paths.sort(key=len,reverse=True)
+
+    for path in pathsOrdered:
+      subpath = list(set(path) - set(CP))
+      subpaths.append(subpath)
+
+    subpathsList = set()
+    for i, subpath in enumerate(subpaths):
+      for j in range(i+1,len(subpaths)):
+        if set(subpath).issubset(subpaths[j]):
+          subpathsList.add()
+        else:
+          pass'''
+
+def expandSubpaths(subpaths, path):
+  expandedPaths = []
+  for subpath in subpaths:
+    idx1 = path.index(subpath[0])
+    if len(subpath)==1:
+      expSubpath = path[idx1-1:idx1+2]
+    else:
+      expSubpath = path[idx1-1:idx1+len(subpath)+1]
+    expandedPaths.append(expSubpath)
+  return expandedPaths
+
+def checkForEndNode(listActivities):
+  for elem in listActivities:
+    if elem.returnName()=='end':
+      return elem
+  return None
+
+def getSubpaths(path,CP):
+  """
+    Method designed to return the set of subpaths that are part of a path parallel to the CP
+    @ In, path, list, list of activities
+    @ In, CP, list, list of activities
```

Comment:
Please provide more details about these two variables, from the docstring, there is no difference between these two variables.